### PR TITLE
Use Doxygen @cite

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

Use Doxygen `@cite` command to create Bibliography.

- The PR creates links for most (?) citations referenced from C++ docstrings.
- Presumably, `sphinxcontrib-bibtex` cannot be used in this context (Doxygen); the `@cite` command is easy to use, and does not require the creation of separate documentation in Sphinx. 
 
See comment in https://github.com/Cantera/enhancements/issues/178#issuecomment-1644858874

Addresses Cantera/enhancements#145 and Cantera/enhancements#179

**If applicable, provide an example illustrating new features this pull request is introducing**

image

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: src/base/application.h [hunk 1]
```diff
@@ -374,17 +374,18 @@ class Application
...
-     * third-body concentrations, consistent with Cantera's behavior prior to version 3.0.
+     * If set to \c false (default value), rate constants of three-body reactions are
+     * consistent with conventional definitions (for example Eq. 9.75 in Kee, et al.
+     * @cite kee2003).
+     * If set to \c true, output for rate constants of three-body reactions is
+     * multiplied by third-body concentrations, consistent with %Cantera's behavior
+     * prior to version 3.0.
      */
-    void use_legacy_rate_constants(bool legacy=true) {
+    void use_legacy_rate_constants(bool legacy=false) {
```
Comment (speth) [inline src/base/application.h]: I don't think this makes sense. Regardless of how the default has changed, calling `use_legacy_rate_constants()` should not _disable_ the legacy rate constant approach.

File: include/cantera/transport/IonGasTransport.h [hunk 2]
```diff
@@ -19,30 +19,14 @@ namespace Cantera
...
- *    Combustion Theory and Modelling 19.6 (2015): 744-772.
- *    DOI: 10.1080/13647830.2015.1090018
- * 4. Chiflikian, R. V. "The analog of Blanc’s law for drift velocities
- *    of electrons in gas mixtures in weakly ionized plasma."
- *    Physics of Plasmas 2.10 (1995): 3902-3909.
- * 5. Viehland, L. A., et al. "Tables of transport collision integrals for
- *    (n, 6, 4) ion-neutral potentials." Atomic Data and Nuclear Data Tables
- *    16.6 (1975): 495-514.
+ * References for Stockmayer-(n,6,4) model: @cite selle1999, @cite selle2000,
+ * @cite han2015, @cite chiflikian1995, @cite viehland1975.
```
Comment (speth) [inline include/cantera/transport/IonGasTransport.h]: Here and in a couple other places, I would recommend we follow the style where the numbered references are not read as part of the sentence structure, e.g.
```suggestion
 * References for Stockmayer-(n,6,4) model: Selle and Riedel @cite selle1999, @cite selle2000; 
 * Han et al. @cite han2015; Chiflikian @cite chiflikian1995; and Viehland et al. @cite viehland1975.
```

**Review by speth — CHANGES_REQUESTED** [review]
Thanks, @ischoegl, I think is an improvement, and may encourage better habits for referencing the literature in the future. I had just a couple of comments.

**Comment by ischoegl** [comment]
Thanks, @speth ... I believe this is taken care of.

**Review by speth — APPROVED** [review]

**Pull request merged**